app/src/main/java/com/example/rhythm/MainActivity.kt

```diff
@@ -2,12 +2,21 @@ package com.example.rhythm
 
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import com.example.rhythm.ui.MusicListFragment
 import dagger.hilt.android.AndroidEntryPoint
 
 @AndroidEntryPoint
-class MainActivity : AppCompatActivity() {
+class MainActivity : AppCompatActivity(R.layout.activity_main) {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_main)
+        launchMusicListFragment()
     }
-}
\ No newline at end of file
+
+    private fun launchMusicListFragment() {
+        val musicListFragment = MusicListFragment.newInstance()
+        with(supportFragmentManager.beginTransaction()) {
```

Comment:
I think we should add a check for saveInstanceState here.. 
```
if(savedInstanceState == null) {
   // commit fragment
}
```
it will avoid adding duplicate fragment on fragment manager, in case of activity is rectreated.

Reply:
Updated, thank you.